src/ota/interface/OTAInterfaceDefault.cpp

```diff
@@ -41,57 +42,43 @@ OTACloudProcessInterface::State OTADefaultCloudProcessInterface::startOTA() {
     }
   );
 
-  // make the http get request
+  // check url
   if(strcmp(context->parsed_url.schema(), "https") == 0) {
     http_client = new HttpClient(*client, context->parsed_url.host(), context->parsed_url.port());
   } else {
     return UrlParseErrorFail;
   }
 
-  http_client->beginRequest();
-  auto res = http_client->get(context->parsed_url.path());
-
-  if(username != nullptr && password != nullptr) {
-    http_client->sendBasicAuth(username, password);
-  }
-
-  http_client->endRequest();
-
-  if(res == HTTP_ERROR_CONNECTION_FAILED) {
-    DEBUG_VERBOSE("OTA ERROR: http client error connecting to server \"%s:%d\"",
-      context->parsed_url.host(), context->parsed_url.port());
-    return ServerConnectErrorFail;
-  } else if(res == HTTP_ERROR_TIMED_OUT) {
-    DEBUG_VERBOSE("OTA ERROR: http client timeout \"%s\"", OTACloudProcessInterface::context->url);
-    return OtaHeaderTimeoutFail;
-  } else if(res != HTTP_SUCCESS) {
-    DEBUG_VERBOSE("OTA ERROR: http client returned %d on  get \"%s\"", res, OTACloudProcessInterface::context->url);
-    return OtaDownloadFail;
-  }
-
-  int statusCode = http_client->responseStatusCode();
-
-  if(statusCode != 200) {
-    DEBUG_VERBOSE("OTA ERROR: get response on \"%s\" returned status %d", OTACloudProcessInterface::context->url, statusCode);
-    return HttpResponseFail;
-  }
+  // make the http get request
+  requestOta(OtaFetchTime);
```

Comment:
The return value is not being handled, an error could be returned

Reply:
fixed here https://github.com/arduino-libraries/ArduinoIoTCloud/pull/520/commits/89dddf7695485a766b54a4f481f35074e3411e37